app/models/abstracts.py

```diff
@@ -7,3 +10,19 @@ class Date(models.Model):
 
     class Meta:
         abstract = True
+
+class SoftDeletionModel(models.Model):
+    is_active = models.BooleanField(default=False)
+
+    objects = SoftDeletionManager()
```

Comment:
If we do the changes in the manager we can remove:
- [ ] the all_objects attr
- [ ] the delete and hard_delete methods
- [ ] we can add a soft_delete method in the manager for more clarity and keep the delete as it is and we'll just use soft delete though the code